docs/src/faq.md

```diff
@@ -0,0 +1,401 @@
+```@meta
+CurrentModule = Enzyme
+DocTestSetup = quote
+    using Enzyme
+end
+```
+
+# Frequently asked questions
+
+## Implementing pullbacks
+
+Enzyme's [`autodiff`](@ref) function can only handle functions with scalar output. To implement pullbacks (back-propagation of gradients/tangents) for array-valued functions, use a mutating function that returns `nothing` and stores it's result in one of the arguments, which must be passed wrapped in a [`Duplicated`](@ref).
```

Comment:
This is only the case in (combined) reverse mode. Split reverse mode (aka autodiff_thunk) and forward mode do handle non-scalar output

Reply:
I'm not the one who wrote this part, I only copy-pasted it from where it was in a specific page to a section of the FAQ

Maybe we save modifications there for another PR?

Reply:
added it